mteb/evaluation/MTEB.py

```diff
@@ -527,8 +532,15 @@ def run(
                     # Determine subsets to run for this split
                     # If the whole split is missing, run all required subsets
                     # If only some subsets are missing, run only those
-                    subsets_to_run = info["missing_subsets"]
-                    if info["whole_split_missing"] and task_subsets is None:
+                    subsets_to_run = (
+                        info["missing_subsets"]
+                        if not overwrite_results
+                        else task_subsets
```

Comment:
Judging from the test failures, maybe this condition needs to be reverted?

Reply:
It seems, that test were incorrect a bit, because they use `overwrite_results=True`

Reply:
`overwrite_results` shouldn't affect how splits are determined, so those tests should pass whether `overwrite_results` is True or False, right?

Reply:
Yes, but in the test, `last_evaluated_splits` was being checked, and with `overwrite_results`, all selected splits and subsets should be reevaluated. This caused the test to fail. For example, instead of just `val` in `last_evaluated_splits`, it should include both `val` and `test`. However, the splits were selected correctly.